Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -101,6 +177,42 @@ public static class ResultExtensions
 		public static Task<Result<TSuccess, TFailure>> Do<TSuccess, TFailure>(this Task<Result<TSuccess, TFailure>> result, Action<TSuccess> success)
 			=> result.Do(success, _ => { });
 
+		public static Result<Option<TSuccess>, TFailure> Do<TSuccess, TFailure>(this Result<Option<TSuccess>, TFailure> result, Action<TSuccess> successSome, Action successNone, Action<TFailure> failure)
+			=> result.Do(x => x.Do(successSome, successNone), failure);
+
+		public static Result<Option<TSuccess>, TFailure> Do<TSuccess, TFailure>(this Result<Option<TSuccess>, TFailure> result, Action<TSuccess> successSome, Action successNone)
+			=> result.Do(x => x.Do(successSome, successNone));
+
+		public static Result<Option<TSuccess>, TFailure> Do<TSuccess, TFailure>(this Result<Option<TSuccess>, TFailure> result, Action<TSuccess> successSome)
```

Comment:
This could collide with the normal Do. You'd need to use explicit types for the generic to do a Do with operates on the Option.

Reply:
Oops, yeah, that should be `DoIfSome`, etc

Reply:
... assuming the `Do`, `Apply` variants are even valuable.  There isn't _that_ much boilerplate being removed by it.